src/test/java/com/yegor256/tojos/MnSynchronizedTest.java

```diff
@@ -0,0 +1,192 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.yegor256.tojos;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link MnSynchronized}.
+ *
+ * @since 0.3.0
+ */
+class MnSynchronizedTest {
+
+    /**
+     * Number of threads.
+     */
+    static final int THREADS = 5;
+
+    /**
+     * The number of changes in under-test mono.
+     */
+    private final AtomicInteger actual = new AtomicInteger(0);
+
+    /**
+     * The mono under test.
+     */
+    private Mono mono;
+
+    /**
+     * The executor.
+     */
+    private ThreadPoolExecutor executor;
+
+    /**
+     * The blocking queue.
+     */
+    private BlockingQueue<Runnable> queue;
+
+    @BeforeEach
+    final void setUp() {
+        this.mono = new MnSynchronized(new MnMemory());
+        this.queue = new LinkedBlockingQueue<>(MnSynchronizedTest.THREADS);
+        this.executor = new ThreadPoolExecutor(
+            MnSynchronizedTest.THREADS,
+            MnSynchronizedTest.THREADS,
+            5L,
+            TimeUnit.SECONDS,
+            this.queue,
+            new ThreadPoolExecutor.CallerRunsPolicy()
+        );
+    }
+
+    /**
+     * Thread-safety test.
+     * In this test, we check the number of changes in MnSynchronized mono.
+     * It should be equal to the sum of the arithmetic progression over the number of threads.
+     *
+     * @throws InterruptedException When fails
+     */
+    @Test
+    final void concurrentScenario() throws InterruptedException {
+        this.executor.prestartAllCoreThreads();
+        for (int trds = 0; trds < MnSynchronizedTest.THREADS; ++trds) {
+            this.queue.offer(
```

Comment:
Use `ExecutorService.invokeAll()` or `ExecutorService.submit()` instead

Reply:
@volodya-lombrozo okay, i'll try to fix